python/tk_desktop/desktop_window.py

```diff
@@ -681,26 +696,24 @@ def __launch_app_proxy_for_project(self, project, pipeline_configuration_id=None
             self.__set_project_just_accessed(project)
             QtGui.QApplication.instance().processEvents()
 
-            if sys.platform == "darwin":
-                path_field = "mac_path"
-            elif sys.platform == "win32":
-                path_field = "windows_path"
-            elif sys.platform.startswith("linux"):
-                path_field = "linux_path"
-            else:
-                raise SystemError("Unsupported platform: %s" % sys.platform)
-
-            filters = [
-                ["project", "is", project],
-            ]
-
-            fields = [path_field, "users", "code"]
+            # Phase 2: Get information about the pipeline configuration.
 
+            # First get the legacy pipeline configurations, i.e. those using the *_path fields.
             connection = engine.shotgun
             pipeline_configurations = connection.find(
                 "PipelineConfiguration",
-                filters,
-                fields=fields,
+                [["project", "is", project]],
+                fields=["mac_path", "windows_path", "linux_path", "users", "code"]
+            )
+            # We don't filter in the Shotgun query for the plugin ids because not every site these fields yet.
```

Comment:
FYI heads up that the sg api handles non-existent fields this way:

```
>>> shotgun.find_one("Project", [], ["name", "field_that_doesnt_exist"])
{'type': 'Project', 'id': 63, 'name': 'Template Project'}
```
They are basically not returned in the return dataset.

(however you cannot use non-existent fields in filters!)